tests/e2e/monitoring/test_when_respin_ceph_pods_interaction_with_prometheus.py

```diff
@@ -0,0 +1,125 @@
+import logging
+
+import pytest
+
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.framework.testlib import tier4, E2ETest
+from tests.fixtures import (
+    create_rbd_storageclass, create_ceph_block_pool,
+    create_rbd_secret
+)
+from tests.helpers import (
+    create_pvc, create_pod,
+    create_project
+)
+from ocs_ci.ocs.monitoring import collected_metrics_for_created_pvc
+from ocs_ci.ocs.resources import pvc, pod
+from tests import disruption_helpers
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def test_fixture(request):
+    """
+    Setup and teardown
+    """
+    self = request.node.cls
+
+    def finalizer():
+        teardown(self)
+    request.addfinalizer(finalizer)
+    setup(self)
+
+
+def setup(self):
+    """
+    Create multiple projects, pvcs and app pods
+    """
+
+    # Initializing
+    self.namespace_list = []
+    self.pvc_objs = []
+    self.pod_objs = []
+
+    assert create_multiple_project_and_pvc_and_check_metrics_are_collected(self)
+
+
+def teardown(self):
+    """
+    Delete app pods and PVCs
+    Delete project
+    """
+    # Delete created app pods and PVCs
+    assert pod.delete_pods(self.pod_objs)
+    assert pvc.delete_pvcs(self.pvc_objs)
+
+    # Switch to default project
+    ret = ocp.switch_to_default_rook_cluster_project()
+    assert ret, 'Failed to switch to default rook cluster project'
+
+    # Delete projects created
+    for prj in self.namespace_list:
+        self.prj_obj.delete(resource_name=prj)
+
+
+def create_multiple_project_and_pvc_and_check_metrics_are_collected(self):
```

Comment:
create_multiple_projects AND create_pvc AND check_metrics_are_collected.
you have 3 functions here - please split

Reply:
done